lib/appsignal/plug.ex

```diff
@@ -78,6 +78,41 @@ if Appsignal.plug? do
       "#{method} #{path}"
     end
 
+    def extract_sample_data(%Plug.Conn{params: params, host: host,
+      method: method, script_name: script_name, request_path: request_path,
+      port: port, query_string: query_string} = conn) do
+
+      %{
+        "params" => Appsignal.Utils.ParamsFilter.filter_values(params),
+        "environment" => %{
+          "host" => host,
+          "method" => method,
+          "script_name" => script_name,
+          "request_path" => request_path,
+          "port" => port,
+          "query_string" => query_string,
+          "request_uri" => url(conn),
+          "peer" => peer(conn)
+        } |> Map.merge(extract_request_headers(conn))
+      }
+    end
+
+    def extract_request_headers(%Plug.Conn{req_headers: req_headers}) do
+      keys = ~w(
+        accept accept-charset accept-encoding accept-language cache-control
+        connection user-agent from referer range
+      )
```

Comment:
Added everything (except for `HTTP_X_MIDDLEWARE_START`) to the whitelist in 41c25c0. Could you open a new issue for `set_queue_start`?